Automattic/wp-calypso
# Domains: Add warning for domain primary change on plan cancellation

#### Changes proposed in this Pull Request

This PR introduces a warning before downgrading to a free plan for users participating in this test: p99Zz8-10j-p2

<img width="555" alt="Screenshot 2020-03-05 at 6 38 57 PM" src="https://user-images.githubusercontent.com/13062352/76008849-9ad53180-5f10-11ea-93b5-bcec28e3dcb9.png">

#### Testing instructions

For a site which has a plan, and at least one registered domain (ideally both the case where a plan is within the refund period, and the case where the plan is outside the refund period):

* When the primary domain is not a registered domain, make sure canceling/refunding subscription works the same, both within and outside the refund period, and that nothing gets broken
* When the primary domain is a registered domain, verify that when canceling a subscription, you get the warning shown in the screenshot above
* Verify that the buttons in the dialog shown do what is expected

Note: if you go through the flow the actual primary domain won't change until the proper backend changes are done
